### PR TITLE
Reduce the test coverage for compatibility against Databricks

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Reduce considerably the compatibility test coverage against the Databricks cluster versions:

- 10.4 LTS
- 11.3 LTS
- 12.2 LTS

This is done out of pragmatical reasons of test runtime and infrastructure cost and it is based on the assumption that there is enough testing done on the oldest and the newest LTS versions of Databricks.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by ebyhr** [comment]
/test-with-secrets sha=cf6733d85a0e8948ad1305d25272d99975b6bc82

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlterTableCompatibility.java [hunk 1]
```diff
@@ -182,7 +185,7 @@ public void testCommentOnTable()
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_DATABRICKS_104, DELTA_LAKE_DATABRICKS_113, DELTA_LAKE_DATABRICKS_122, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlterTableCompatibility.java:188]: The condition of when we should run all is unclear to me.

1. The purpose of this `testCommentOnTableUnsupportedWriterVersion` is ensuring the failure of new writer version, so I thought running on the newest LTS version is sufficient. 
2. When we see this entire class, I wonder why it covers only COMMENT ON COLUMN statement.

**Comment by hashhar** [comment]
It might make it easier to understand and apply the logic going forward if all the tests which need to run across all versions are in a separate test class.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlterTableCompatibility.java (hunk 1 above)
Comment (findinpath) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlterTableCompatibility.java]: Removed the testing from TestDeltaLakeAlterTableCompatibility.

Comment (findinpath) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlterTableCompatibility.java]: > The condition of when we should run all is unclear to me.

+1 

We are currently running a reduced set of tests < 10 minutes to see whether there is anything obvious failing.
And also to make sure that the product test environment for the Databricks version 10.x , 11.x, 12.x doesn't rot.

**Comment by findinpath** [comment]
> It might make it easier to understand and apply the logic going forward if all the tests which need to run across all versions are in a separate test class.

If the set of tests executed on all the clusters increases, the class grouping all these tests will not be that readable anymore.
In case we just run a very, very reduced set of tests, then yes, we could group everything under one class.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteDeltaLakeDatabricks104.java [hunk 2]
```diff
@@ -31,7 +31,7 @@ public List<SuiteTestRun> getTestRuns(EnvironmentConfig config)
     {
         return ImmutableList.of(
                 testOnEnvironment(EnvSinglenodeDeltaLakeDatabricks104.class)
-                        .withGroups("configured_features", "delta-lake-databricks")
+                        .withGroups("configured_features", "delta-lake-databricks-104")
                         .withExcludedGroups("delta-lake-exclude-104")
```
Comment (pajaks) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteDeltaLakeDatabricks104.java:35]: Do we need to have exclude if we need to explicitly add test to `delta-lake-databricks-104` group?

**Comment by findinpath** [comment]
Addressed comments.

**Review by pajaks — APPROVED** [review]

**Comment by ebyhr** [comment]
/test-with-secrets sha=af88d9bfb958183b9fcb7a4df10426089af07c14

**Review by ebyhr — APPROVED** [review]
The following block in `TestDeltaLakeDatabricksCreateTableCompatibility#getDatabricksDefaultTableProperties` looks redundant now:
```java
        if (databricksRuntimeVersion.equals(DATABRICKS_104_RUNTIME_VERSION)) {
            return "TBLPROPERTIES (\n" +
                    "  'Type' = 'EXTERNAL',\n" +
                    "  'delta.minReaderVersion' = '1',\n" +
                    "  'delta.minWriterVersion' = '2')\n";
        }
```

**Pull request merged**